### ISSUE TITLE
Problems in which descriptions of component blocks are not translated.

### ISSUE THREAD

**Issue by semicolon3253** [mit-cml/appinventor-sources#2624]
**Describe the bug**

![image](https://user-images.githubusercontent.com/47348424/147868344-f15ee8bb-af73-4161-ac45-eee7ec3bb3cf.png)
As shown in the picture above, the description of the component's events, procedures, and property blocks is displayed in English regardless of the set language.
This appears only in blocks of components(buttons, etc.) and screen, and does not appear in built-in blocks such as control, logic, etc.

**Affects**

- [ ] Designer
- [x] Blocks editor
- [ ] Companion
- [ ] Compiled apps
- [ ] Buildserver
- [ ] Debugging
- [ ] Other... (please describe)

**Expected behavior**

It can be difficult for users unfamiliar with English to understand the blocks.

**Steps to reproduce**

You can just access the appinventer and check it right away.

I did an experiment to see if this problem actually does not apply to translation files (OdeMessages_xx.properties).
![image](https://user-images.githubusercontent.com/47348424/147868533-5dfd993a-8515-4493-ada0-4c2563b5a50a.png)
For example, I changed the description of the Timer event to "Description in Clock.java" in file Clock.java, the source code of the clock component, and built it to check.
![image](https://user-images.githubusercontent.com/47348424/147868676-d4313ea4-5152-44b9-8b32-c09f0a794d18.png)
Then, as in the picture above, the revised description regardless of language was displayed as it was.

I don't think the translation file is applied normally. But I don't know where the problem is.

**Comment by semicolon3253** [comment]
I found the reason for the problem.
Among the keys in the translation string, the key of the string related to 'description' seems to have changed from before.
The keys of the string related to the description of the event, method, and property were added with 'EventDescriptions', 'MethodDescriptions', and 'PropertyDescriptions' after the key, respectively(Example, Timer**EventDescriptions**).
However, the source code was not updated in response to this, so the corresponding key was not found and the untranslated string was returned.

This is the source codes where the problem occurred:
https://github.com/mit-cml/appinventor-sources/blob/981ec16d89c49b5cba5f2aff828a9d609b135a23/appinventor/blocklyeditor/src/component_database.js#L782-L790
https://github.com/mit-cml/appinventor-sources/blob/981ec16d89c49b5cba5f2aff828a9d609b135a23/appinventor/blocklyeditor/src/component_database.js#L802-L810
https://github.com/mit-cml/appinventor-sources/blob/981ec16d89c49b5cba5f2aff828a9d609b135a23/appinventor/blocklyeditor/src/component_database.js#L832-L840

The above functions use the key of the translation(parameter component and name) to get the translation of the matching key from 'i18n*Descriptions_', the array in which the translated sentences are stored.
However, for the sentences related to the description, the above-mentioned prefix does not exist in the parameter 'name', so the value cannot be found and the untranslated sentence 'opt_default' is returned.

I revised component_database.js as below to solve this problem.
```javascript
return this.i18nEventDescriptions_[component + '.' + name + 'EventDescriptions'] || this.i18nEventDescriptions_[name + 'EventDescriptions'] || opt_default || name;
```
Simply put a prefix after the key when bringing in sentences related to the explanation.
Except for some languages that have not been translated, the problem was solved in this way.
Languages without translation still come out in English, but this is not a matter of code, but it has not been translated, so we just have to wait for the translation to be completed.
I'd like to hear your opinion on this, so I'll leave this issue open for a while before I make a PR.
